src/unix/linux-core.c

```diff
@@ -281,11 +285,18 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     if (sizeof(int32_t) == sizeof(long) && timeout >= max_safe_timeout)
       timeout = max_safe_timeout;
 
+#if !(defined(__ANDROID_API__) && __ANDROID_API__ < 21)
     nfds = epoll_pwait(loop->backend_fd,
                        events,
                        ARRAY_SIZE(events),
                        timeout,
                        psigset);
+#else
+    nfds = epoll_wait(loop->backend_fd,
+                      events,
+                      ARRAY_SIZE(events),
+                      timeout);
```

Comment:
This is semantically incorrect because it ignores `psigset`.

Libuv only conditionally used `epoll_wait()` until recently, see commit 9208df0c93 - the `pthread_sigmask()` calls in particular.

The logic here should be something like this:
```c
if (psigset != NULL)
  if (pthread_sigmask(SIG_BLOCK, psigset, NULL))
    abort();

nfds = epoll_wait(...);

if (psigset != NULL)
  if (pthread_sigmask(SIG_UNBLOCK, psigset, NULL))
    abort();
```

I'd be open to reverting the aforementioned commit if that makes things easier for Android but there was an issue on MIPS that I think was caused by passing the wrong sigmask size to the kernel. (On some architectures it's not always `_NSIG / 8` or `sizeof(sigset_t)`, apparently...)

Reply:
I reverted this commit f43c66343316878da09949daf49b515249f71bdf, it seems to solve the problem of epoll_wait.